libgnucash/engine/gnc-option-date.cpp

```diff
@@ -551,7 +533,31 @@ gnc_relative_date_to_time64(RelativeDatePeriod period)
             else if (reldate_is_next(period))
                 now.tm_mday += 7;
     }
-    reldate_set_day_and_time(now, checked_reldate(period).m_type);
+
+    switch (checked_reldate(period).m_type)
+    {
+        case RelativeDateType::ABSOLUTE:
+	case RelativeDateType::LAST:
+        case RelativeDateType::NEXT:
+            // Do nothing
+            break;
+        case RelativeDateType::START:
+            now.tm_mday = offset == RelativeDateOffset::QUARTER && acct_per.tm_mday > 1 ? acct_per.tm_mday : 1;
+            gnc_tm_set_day_start(&now);
+            break;
+        case RelativeDateType::END:
+        {
+            /* Ensure that the month is between 0 and 11*/
+            auto year_delta = (now.tm_mon / 12) + (now.tm_mon < 0 ? -1 : 0);
+            auto month = now.tm_mon - (12 * year_delta);
+            auto year = now.tm_year + year_delta;
+            now.tm_mday = days_in_month(month, year);
+            if (offset == RelativeDateOffset::QUARTER && acct_per.tm_mday > 1 && acct_per.tm_mday <= now.tm_mday)
+                now.tm_mday = acct_per.tm_mday - 1;
```

Comment:
Doesn't cover quarters ending in February when the  accounting period begins on the 30th or 31st of a longer month.

Reply:
It should now.